plugins/ltac2/tac2intern.ml

```diff
@@ -1072,6 +1072,12 @@ let tycon_app ?loc env ~ft t =
         Pp.(str "This expression has type" ++ spc() ++ pr_glbtype env ft ++ str"." ++
             spc() ++ str "It is not a function and cannot be applied.")
 
+let warn_useless_record_with = CWarnings.create ~name:"ltac2-useless-record-with"
```

Comment:
Can we get an error by default? The user can always turn this off when dealing with compat code.